websocket/jetty10/src/main/java/jenkins/websocket/Jetty10Provider.java

```diff
@@ -44,14 +46,34 @@
 @MetaInfServices(Provider.class)
 public class Jetty10Provider implements Provider {
 
+    /**
+     * Number of seconds a WebsocketConnection may stay idle until it expires.
+     * Zero to disable.
+     * This value must be higher than the <code>jenkins.websocket.pingInterval</code>.
+     * Per <a href=https://www.eclipse.org/jetty/documentation/jetty-10/programming-guide/index.html#pg-websocket-session-ping>Jetty 10 documentation</a>
+     * a ping mechanism should keep the websocket active. Therefore, the idle timeout must be higher than the ping
+     * interval to avoid timeout issues.
+     */
+    private static long IDLE_TIMEOUT_SECONDS = Long.getLong("jenkins.websocket.idleTimeout", 60L);
+
     private static final String ATTR_LISTENER = Jetty10Provider.class.getName() + ".listener";
 
+    private final AtomicBoolean initialized = new AtomicBoolean(false);
+
     public Jetty10Provider() {
         JettyWebSocketServerContainer.class.hashCode();
     }
 
+    private void init(HttpServletRequest req) {
+        if (!initialized.get()) {
+            JettyWebSocketServerContainer.getContainer(req.getServletContext()).setIdleTimeout(Duration.ofSeconds(IDLE_TIMEOUT_SECONDS));
+            initialized.set(true);
```

Comment:
This is not actually atomic—you might as well use a simple `boolean`. I doubt it matters since calling `setIdleTimeout` twice should be harmless.